components/calendar/src/provider.rs

```diff
@@ -151,3 +152,389 @@ pub struct WeekDataV1 {
+        WeekendDays {
+            first_weekday,
+            current_day: first_weekday.next_day(),
+            weekend,
+        }
+    }
+}
+
+impl Iterator for WeekendDays {
+    type Item = IsoWeekday;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        // Check each bit until we find one that is ON or until we are back to the start of the week.
+        while self.current_day != self.first_weekday {
+            if self.weekend.0 & self.current_day.bit_value() != 0 {
+                let result = self.current_day;
+                self.current_day = self.current_day.next_day();
+                return Some(result);
+            } else {
+                self.current_day = self.current_day.next_day();
+            }
+        }
+
+        if self.weekend.0 & self.current_day.bit_value() != 0 {
+            // Clear weekend, we've seen all bits.
+            // Breaks the loop next time `next()` is called
+            self.weekend = WeekendSet(0);
+            return Some(self.current_day);
+        }
+
+        Option::None
+    }
+}
+
+/// An ICU4X mapping to a subset of CLDR weekData.
+/// See CLDR-JSON's weekData.json for more context.
+///
+/// <div class="stab unstable">
+/// 🚧 This code is considered unstable; it may change at any time, in breaking or non-breaking ways,
+/// including in SemVer minor releases. While the serde representation of data structs is guaranteed
+/// to be stable, their Rust representation might not be. Use with caution.
+/// </div>
+#[icu_provider::data_struct(marker(
+    WeekDataV2Marker,
+    "datetime/week_data@2",
+    fallback_by = "region"
+))]
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[cfg_attr(feature = "datagen", derive(serde::Serialize))]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[allow(clippy::exhaustive_structs)] // used in data provider
+pub struct WeekDataV2 {
+    /// The first day of a week.
+    pub first_weekday: IsoWeekday,
+    /// For a given week, the minimum number of that week's days present in a given month or year for the week to be considered part of that month or year.
+    pub min_week_days: u8,
+    /// Bitset representing weekdays that are part of the 'weekend', for calendar purposes.
+    /// The number of days can be different between locales, and may not be contiguous.
+    weekend_set: WeekendSet,
+}
+
+impl WeekDataV2 {
+    /// Constructs Week Data, including an internal representation of the weekend.
+    pub fn new(first_weekday: IsoWeekday, min_week_days: u8, weekend_set: WeekendSet) -> Self {
+        WeekDataV2 {
+            first_weekday,
+            min_week_days,
+            weekend_set,
+        }
+    }
+
+    /// Constructs Week Data, including an internal representation of the weekend. Used from compiled data.
+    pub const fn from_baked(first_weekday: u8, min_week_days: u8, weekend_set: WeekendSet) -> Self {
+        WeekDataV2 {
+            first_weekday: IsoWeekday::from_usize(first_weekday as usize),
```

Comment:
why go through `usize`? `new` looks like the constructor you want to bake.

Reply:
Otherwise [here](https://github.com/JhonnyBillM/icu4x/blob/e3b3aed577197cbf361022ffccf632e3d7692855/components/calendar/src/provider.rs#L430) I cannot get the token for `IsoWeekday`.
I tried with `self.first_weekday.bake(ctx)` but it doesn't have it (unexpected for me because IsoWeekday derives `Baked`).

Open to suggestions on how to handle this

Reply:
What's the error? I don't see why `self.first_weekday.bake(ctx)` shouldn't work